aws/src/main/java/org/apache/iceberg/aws/s3/S3OutputStream.java

```diff
@@ -212,9 +212,7 @@ public void write(byte[] b, int off, int len) throws IOException {
   }
 
   private void newStream() throws IOException {
-    if (stream != null) {
-      stream.close();
-    }
+    closeStream();
```

Comment:
This change seems unrelated and I'm not sure what it really adds.  It looks like we're dereferencing the stream, but we still would throw in any situation where it actually have any affect.

Reply:
This will always throw on an aborted stream on a checksum check failure, so it could be very noisy.

Reply:
`stream.close()` was being called without the null check in one place, so this was added to prevent a potential NPE, however this is unrelated to the performance regression.

Reply:
I rolled this back